### PR TITLE
Fetch Data: Fix Display Bugs

### LINKED ISSUE #80: Rewrite findPantryInfo() in scripts.js
- [ ] Need to rewrite findPantryInfo in scripts.js

- [ ] Determine whether pantryInfo should be displayed an array or an array of objects?

### LINKED ISSUE #83: Bug: API RecipeData does not have ingredient names :-(
API Recipe Ingredient:
{
          "id": 20081,
          "quantity": {
            "amount": 1.5,
            "unit": "c"
          }

vs
Repo Data Recipe Ingredient:
      {
        "name": "vanilla extract",
        "id": 2050,
        "quantity": {
          "amount": 0.5,
          "unit": "tsp"
        }
      }

### LINKED ISSUE #84: Bug: Pantry Ingredients does not have any way to cross reference with selected ingredients
- [ ] My Pantry drop down has many checkboxes and references with them by name, however, the API only references with the id instead of the name.

### LINKED ISSUE #90: Refactor: Find scripts functions that need variables changed to account for API data use

### PR DESCRIPTION
**Pull request by hannakim91**
### What’s this PR do?  

- Lots of small bug fixes - mainly renaming variables used in different display methods to use fetched API data.
- Also add code to generateIngredients to get ingredient names into each recipe's ingredients --- likely needs to be moved to a class (recipe)
- Display works up until being able to favorite/unfavorite to save to my recipes: Thought we fixed image bug (with images in repo) displaying but that is still not complete -- doesn't show again after restarting server// also may have to do with how API data is getting fetched?
- Fix indent errors in webpack.config.js
### Where should the reviewer start?  
- scripts.js
- webpack.config.js
- index.html
### How should this be manually tested?  
run `npm start` to start server and open on browser: http://localhost:8080/
Any background context you want to provide?  

### What are the relevant tickets?  

- Resolves #80 
- Resolves #83 
- Resolves #84 
- Resolves #90

### Questions: 

- We discussed creating a `recipeRepo`class
- How to organize `fetch.js` and dom display functions

### CODE REVIEW

File: src/fetch.js [hunk 1]
```diff
@@ -1,3 +1,8 @@
+//fetching 3 times, then promise.all resolves redundancies -- storing fetched data as obj properties(allData)
+//return in allData?
+
+// instead: start with Promise.all, two thens, and a catch
+
```
Comment (JWoltjen) [inline src/fetch.js:5]: Could consider refactoring fetch.js but it works right now.

File: src/index.html [hunk 2]
```diff
@@ -3,21 +3,21 @@
...
+      <h1><span>fresh</span>Picks<img src="./images/apple-logo.png" alt="apple icon" class="header-apple-icon"></h1>
       <form id="search" role="search">
         <input id="search-input" type="text" autocomplete="off" placeholder="Search Recipes..."></input>
         <label for="search-input" class="search-label">Search</label>
-        <button class="nav-btn search-btn" type="button" name="button"><img src="../images/search.png" alt="search button"></button>
+        <button class="nav-btn search-btn" type="button" name="button"><img src="./images/search.png" alt="search button"></button>
       </form>
-      <button class="nav-btn ing-pan-btn saved-recipes-btn" type="button" name="button"><img src="../images/cookbook.png" alt="search button">My Recipes</button>
-      <button class="nav-btn my-pantry-btn ing-pan-btn saved-ingredients-btn" type="button" name="button"><img src="../images/seasoning.png" alt="search button">My Pantry</button>
+      <button class="nav-btn ing-pan-btn saved-recipes-btn" type="button" name="button"><img src="./images/cookbook.png" alt="search button">My Recipes</button>
```
Comment (aemcdonald) [inline src/index.html:19]: Great job fixing the files paths on these images!

File: src/scripts.js [hunk 3]
```diff
@@ -66,23 +70,24 @@ function generateUser() {
...
     if (recipeInfo.name.length > 40) {
       shortRecipeName = recipeInfo.name.substring(0, 40) + "...";
     }
-    addToDom(recipeInfo, shortRecipeName)
+    displayRecipeDetails(recipeInfo, shortRecipeName)
   });
 }
 
-function addToDom(recipeInfo, shortRecipeName) {
+function displayRecipeDetails(recipeInfo, shortRecipeName) {
```
Comment (aemcdonald) [inline src/scripts.js:90]: Great idea changing the name of this function to be more semantic.

File: src/scripts.js [hunk 4]
```diff
@@ -231,8 +237,17 @@ function addRecipeImage(recipe) {
   document.getElementById("recipe-title").style.backgroundImage = `url(${recipe.image})`;
 }
 
+// ********
 function generateIngredients(recipe) {
```
Comment (aemcdonald) [inline src/scripts.js:241]: I think we should consider moving this function into the recipe class to keep data manipulation separate from displaying on/updating the DOM

File: src/fetch.js (hunk 1 above)
Comment (npdarrington) [reply src/fetch.js:5]: Agreed with Jeff on this one. Setting up the fetch.js file this way feels too dependant on the fetchData call and makes our calls more restrictive.

File: src/index.html [hunk 5]
```diff
@@ -3,21 +3,21 @@
   <head>
     <meta charset="utf-8">
     <title>Fresh Picks</title>
-    <link rel="stylesheet" type="text/css" href="../styles.css">
+    <!-- <link rel="stylesheet" type="text/css" href="../styles.css"> -->
```
Comment (npdarrington) [inline src/index.html:6]: This styles.css file is indeed not doing anything. Further research on webpack shows that webpack does not crawl through HTML files, meaning most of our files will link into our scripts.js file. Webpack actually does not see this file.

File: src/scripts.js [hunk 6]
```diff
@@ -1,4 +1,5 @@
 import './css/styles.scss';
+// import './images/apple-logo-png'
```
Comment (npdarrington) [inline src/scripts.js:2]: Let's load all of our images into the index.js file and link that inside. This does work how we linked the file here, we just have to run `npm run build` in order for it to build the dist folder where webpack will now find the files properly.

File: src/scripts.js (hunk 4 above)
Comment (npdarrington) [reply src/scripts.js:241]: Agreed with Ashley on splitting our DOM updates into separate files and linking the file into scripts.js.

File: src/scripts.js [hunk 7]
```diff
@@ -231,8 +237,17 @@ function addRecipeImage(recipe) {
...
+  const mappedRecipe = recipe.ingredients.map(recipeIngredient => {
+    ingredientsData.forEach(ingredient => {
+      if (ingredient.id === recipeIngredient.id) {
+        recipeIngredient.name = ingredient.name
+      }
+    })
+    return recipeIngredient
+  })
+  return recipe && mappedRecipe.map(i => {
     return `${capitalize(i.name)} (${i.quantity.amount} ${i.quantity.unit})`
```
Comment (npdarrington) [inline src/scripts.js:251]: Great call on this.

**Review by npdarrington — APPROVED** [review]

File: src/index.js [hunk 8]
```diff
@@ -1,13 +1,13 @@
...
 // An example of how you import jQuery into a JS file if you use jQuery in that file
 import $ from 'jquery';
 
 // An example of how you tell webpack to use a CSS (SCSS) file
 import './css/base.scss';
 
 // An example of how you tell webpack to use an image (also need to link to it in the index.html)
-import './images/turing-logo.png'
+import './images/apple-logo.png'
+import './images/search.png'
```
Comment (JWoltjen) [inline src/index.js:11]: Do we need to load all the images for webpack to pick them up? Not clear on this yet.

File: src/scripts.js (hunk 6 above)
Comment (JWoltjen) [reply src/scripts.js:2]: I agree with Nathan, I think.

**Review by aemcdonald — APPROVED** [review]
Excellent job on all the bug fixes. I think we should consider moving generateIngredients into the recipe class to separate data manipulation from DOM manipulation.

**Review by JWoltjen — APPROVED** [review]
Great job, Hannah!

**Pull request merged**